subsys/logging/backends/log_backend_net.c

```diff
@@ -48,12 +46,33 @@ static int line_out(uint8_t *data, size_t length, void *output_ctx)
 {
 	struct log_backend_net_ctx *ctx = (struct log_backend_net_ctx *)output_ctx;
 	int ret = -ENOMEM;
+	struct msghdr msg = { 0 };
+	struct iovec io_vector[2];
+	int pos = 0;
 
 	if (ctx == NULL) {
 		return length;
 	}
 
-	ret = zsock_send(ctx->sock, data, length, ZSOCK_MSG_DONTWAIT);
+#if defined(CONFIG_NET_TCP)
+	char len[sizeof("123456789")];
+
+	if (ctx->is_tcp) {
```

Comment:
Could this be a runtime check, ie `IS_ENABLED(CONFIG_NET_TCP) && ctx->is_tcp`?

Reply:
Was this overlooked in the recent push?

Reply:
There is the `CONFIG_NET_TCP` few lines above but I can certainly put those inside `IS_ENABLED`, dunno why I did it like that.

Reply:
Remembered now why it was done without IS_ENABLED, the len var is allocated from stack so it cannot be inside a block

Reply:
Hmm, I just wonder if it wouldn't be compiled out if not used, even if declared outside of the block. But in the end, it's nothing important so we can leave as is as well.

Reply:
If using IS_ENABLED() we would referece invalid pointer in sendmsg() call